Store and Retrieve Data: Quick Start Page & Get Started Page (#4)
datahaven-xyz/datahaven-docs: store-and-retrieve-data/use-storagehub-sdk/get-started.md

---
title: Get Started

> description: Get up and running quickly with the StorageHub SDK.

Reviewer: ```suggestion
description: TODO
```
What does up and running mean? Am I connecting to a network? Sending some kind of request? This title doesn't tell me and the page currently doesn't answer this either.

slug: /store-and-retrieve-data/use-storagehub-sdk/get-started
---

# Get Started with the StorageHub SDK

## Overview

> The StorageHub SDK is a developer toolkit that makes it easy to build on DataHaven by providing direct access to StorageHub pallet functionalities for managing storage, buckets, and proofs.

Reviewer: This page spends a lot of words telling me what and very few telling me how. Most of this info can probably be moved to your Overview page for the StorageHub Sdk. This page should be like:

-> What is this thing? Your first sentence here answers that well. 
-> Do I need to make a decision about which packages to install, depending upon what I'm building, or will I always install both of these packages? If I will always install both, all the information on what's in the two packages and their use cases, etc., can be moved to the Overview.  Only include the info here if I need to make decisions on which to install.
-> The part about the MSP thing is confusing for me. There is no way to build one out right now, even if I wanted to, so mentioning that doesn't make sense here. 
-> Prerequisites is generally good but I'll have some formatting specific suggestions I'll leave on those lines
-> "Within your desired project, install the SDK dependencies using the following commands:" is like telling me "Now press down the button" as step #1 to make toast. Let's make step #1 something like "In your desired directory, create a new TypeScript project with the following command:" and take them step by step from there. 
-> Being able to tell if they've done something the right way is super reassuring for users. If I run those install commands, what does successful terminal ouput look like so I know it worked?

@storagehub-sdk/core

- Backend‑agnostic building blocks (wallets, EIP‑1193, precompile helpers bridging Substrate↔EVM, Merkle/WASM utilities, HttpClient, shared types).
- Signing, Merkle/proofs, precompile calls, low‑level HTTP, shared types.
- This package includes EVM account‑typed helpers, WASM‑backed file utilities, and stable primitives **usable without any backend**.
- [Read more](https://www.npmjs.com/package/@storagehub-sdk/core){taget=_blank}

@storagehub-sdk/msp-client

- Retrieving MSP‑specific client information (health, auth nonce/verify, upload/download endpoints). All MSP‑tied logic lives here.
- Talking to an MSP backend (auth + file transfer).
- Includes REST contracts for MSP, token handling, streaming/multipart upload and download helpers.
- [Read more](https://www.npmjs.com/package/@storagehub-sdk/msp-client){target=_blank}

## MSP Service Endpoints

In order to interact with a Main Storage Provider (MSP) through `@storagehub-sdk/msp-client`, a service endpoint is required. A service endpoint is comprised of:

> - A running MSP backend:

Reviewer: ```suggestion
```

>   - A production MSP service endpoint, or

Reviewer: ```suggestion
```

>   - A local MSP backend for development/testing

Reviewer: ```suggestion
```

- StorageHub node 
  - The MSP backend requires connection to a StorageHub blockchain node

Instead of building this out yourself you can just use any of the following MSP base urls for storing and retrieving your files:

 |      Network      |                                   Value                                   |
 |:-----------------:|:-------------------------------------------------------------------------:|
 | DataHaven Testnet | <pre>```https://deo-dh-backend.stagenet.datahaven-infra.network/```</pre> |
                    

## Prerequisites

System requirements

- Node.js ≥ 18 (recommended LTS version)

> - pnpm, npm, or yarn for package management

Reviewer: ```suggestion
- pnpm, npm, or yarn installed for package management
```

> Blockchain access

Reviewer: ```suggestion
```

- Network endpoints (They can be found in the [Quickstart page](/store-and-retrieve-data/quick-start.md){target=_blank})
    - RPC URL
    - Web Socket URL 

> - Test tokens (They can be found in the [Quickstart page](/store-and-retrieve-data/quick-start.md){target=_blank})

Reviewer: ```suggestion
- [Test tokens](TODO: link them to the faucet){target=_blank})
```
link them to the faucet

> 

Reviewer: ```suggestion
```

MSP access

> 

Reviewer: ```suggestion
```

- Backend‑agnostic building blocks (wallets, EIP‑1193, precompile helpers bridging Substrate↔EVM, Merkle/WASM utilities, HttpClient, shared types).
- Signing, Merkle/proofs, precompile calls, low‑level HTTP, shared types.
- This package includes EVM account‑typed helpers, WASM‑backed file utilities, and stable primitives **usable without any backend**.

[@storagehub-sdk/msp-client](https://www.npmjs.com/package/@storagehub-sdk/msp-client){target=_blank}

- Retrieve MSP‑specific client information (health, auth nonce/verify, upload/download endpoints). All MSP‑tied logic lives here.
- Talk to an MSP backend (auth + file transfer).
- Includes REST contracts for MSP, token handling, streaming/multipart upload and download helpers.

## Prerequisites

Before you begin, ensure you have the following:

- [Node.js ≥ 22](https://nodejs.org/en/download) (recommended LTS version) installed  
- [pnpm](https://pnpm.io/){taget=_blank}, [npm](https://www.npmjs.com/){target=_blank}, or [yarn](https://yarnpkg.com/){target=_blank} installed for package management
- [RPC endpoint URL](#configure-network)
- [Web Socket URL](#configure-network)
- [Test tokens](TODO: link to the faucet){target=_blank}

> - [MSP base url](#get-msp-url)

Reviewer 2: ```suggestion
- [MSP base URL](#get-msp-url)
```

---
title: Get Started
description: TODO
slug: /store-and-retrieve-data/use-storagehub-sdk/get-started
---

# Get Started with the StorageHub SDK

## Overview

The StorageHub SDK is a developer toolkit that makes it easy to build on DataHaven by providing direct access to StorageHub pallet functionalities for managing storage, buckets, and proofs.

> The SDK is split into two packages for convenient separation of concerns: `@storagehub-sdk/core` and `@storagehub-sdk/msp-client`. Both work in browser and Node.js environments. You may not always need both at once, so it’s useful to understand the purpose and typical use cases of each:

Reviewer 2: It would be great to have a quick one-liner in here that distinguishes the two packages in one sentence, for example:

As a quick rule of thumb, use @storagehub-sdk/core for everything chain-facing (keys, precompiles, proofs), and @storagehub-sdk/msp-client for everything provider-facing (MSP login, buckets, file uploads/downloads).

@storagehub-sdk/core

- Backend‑agnostic building blocks (wallets, EIP‑1193, precompile helpers bridging Substrate↔EVM, Merkle/WASM utilities, HttpClient, shared types).
- Signing, Merkle/proofs, precompile calls, low‑level HTTP, shared types.
- This package includes EVM account‑typed helpers, WASM‑backed file utilities, and stable primitives **usable without any backend**.

> - [Read more](https://www.npmjs.com/package/@storagehub-sdk/core){target=_blank}

Reviewer 2: I would move the read more Bullet link to @storagehub-sdk/core

and remove Read More from the bulleted list

@storagehub-sdk/msp-client

- Retrieving MSP‑specific client information (health, auth nonce/verify, upload/download endpoints). All MSP‑tied logic lives here.
- Talking to an MSP backend (auth + file transfer).
- Includes REST contracts for MSP, token handling, streaming/multipart upload and download helpers.

> - [Read more](https://www.npmjs.com/package/@storagehub-sdk/msp-client){target=_blank}

Reviewer 2: Same for this I would move the link to @storagehub-sdk/msp-client
and then remove read more from the bullet list

## MSP Service Endpoints

To interact with a Main Storage Provider (MSP) through `@storagehub-sdk/msp-client`, a service endpoint is required. You can use any of the following MSP base urls as an endpoint for storing and retrieving your files:

 |      Network      |                                   Value                                   |
 |:-----------------:|:-------------------------------------------------------------------------:|
 | DataHaven Testnet | <pre>```https://deo-dh-backend.stagenet.datahaven-infra.network/```</pre> |
                    

## Prerequisites

Before you begin, ensure you have the following:

> - Node.js ≥ 18 (recommended LTS version) installed

Reviewer 2: ```suggestion
- [Node.js ≥ 18](https://nodejs.org/en/download) (recommended LTS version) installed
```

> - pnpm, npm, or yarn installed for package management

Reviewer 2: You can link to pnpm, npm, and yarn for reader convenience

- [RPC endpoint URL](TODO link to that section in Quickstart)
- [Web Socket URL](TODO link to that section in Quickstart)
- [Test tokens](TODO: link them to the faucet){target=_blank}
- [MSP base url](TODO)

## Create a New Project Folder

To create a new project folder, execute the following command in the terminal:

```shell
mkdir datahaven-project && cd datahaven-project
```

## Initialize a package.json file

To initialize a package.json file, execute the following command in the terminal:

=== "pnpm"

    ```shell
    pnpm init
    ```

=== "yarn"

    ```shell
    yarn init
    ```

=== "npm"

    ```shell

>     npm init

Reviewer 2: ```suggestion
    npm init --y
```

    ```

The output of that command should be something like this:

```text
Wrote to .../datahaven-project/package.json

{
  "name": "datahaven-project",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "packageManager": "pnpm@10.17.0"
}
```

## Add TypeScript and Node type definitions

To add the TypeScript and Node type definitions to your projects, execute the following command in the terminal:

```shell

> pnpm add -D typescript ts-node @types/node

Reviewer 2: Extra pnpm command here, you can delete this one
